[bug] in Resume for Component

```
component/config.go:43:31: cannot use c.xmpp (type *xmpp.Component) as type xmpp.StreamClient in argument to xmpp.NewStreamManager:
	*xmpp.Component does not implement xmpp.StreamClient (wrong type for Resume method)
		have Resume() error
		want Resume(xmpp.SMState) error
```

--- a/component.go
+++ b/component.go
@@ -108,7 +108,7 @@ func (c *Component) Connect() error {
 	}
 }
 
-func (c *Component) Resume() error {
+func (c *Component) Resume(SMState) error {
 	return errors.New("components do not support stream management")
 }
 

--- a/component_test.go
+++ b/component_test.go
@@ -23,3 +23,10 @@ func TestHandshake(t *testing.T) {
 func TestGenerateHandshake(t *testing.T) {
 	// TODO
 }
+
+// Test that NewStreamManager can accept a Component.
+//
+// This validates that Component conforms to StreamClient interface.
+func TestStreamManager(t *testing.T) {
+	NewStreamManager(&Component{}, nil)
+}

